src/excmds.ts

```diff
@@ -2375,30 +2375,58 @@ export async function tabclosealltoleft() {
     browser.tabs.remove(ids)
 }
 
-/** restore most recently closed tab in this window unless the most recently closed item was a window */
+/** Restore the most recently closed item.
+    The default behaviour is to restore the most recently closed tab in the
+    current window unless the most recently closed item is a window.
+
+    Supplying either "tab" or "window" as an argument will specifically only
+    restore an item of the specified type.
+
+    @param item
+        The type of item to restore. Valid inputs are "recent", "tab" and "window".
+    @return
+        The tab or window id of the restored item. Returns -1 if no items are found.
+ */
 //#background
-export async function undo() {
+export async function undo(item = "recent"): Promise<Number> {
     const current_win_id: number = (await browser.windows.getCurrent()).id
     const sessions = await browser.sessions.getRecentlyClosed({ maxResults: 10 })
```

Comment:
Is returning more than 10 sessions expensive? It doesn't seem to be on my computer and the commit that added this limit seemed to only care about making the code typescript-compliant, so we could probably remove `maxResults: 10` and be fine.

Reply:
I didn't explicitly check it on my machine, I just reused what was already there. I'm fine with removing it but I'm not sure what it defaults to. 

The "Recently" in `browser.sessions.getRecentlyClosed()` would imply that there is an upper limit of sessions retreived.

Reply:
The upper limit is currently 25 (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sessions/MAX_SESSION_RESULTS).

Reply:
That seems totally reasonable. Not sure why the limit of 10 was added.